vsintegration/src/FSharp.Editor/LanguageService/FSharpAnalysisSaveFileCommandHandler.fs

```diff
@@ -45,43 +46,41 @@ type internal FSharpAnalysisSaveFileCommandHandler [<ImportingConstructor>] (ana
-
-                                            openDocIds
-                                            |> Seq.filter (fun x ->
-                                                depProjIds.Contains(x.ProjectId)
-                                                && x <> document.Id
-                                                && (let doc = solution.GetDocument(x)
-
-                                                    match box doc with
-                                                    | null -> false
-                                                    | _ -> doc.Project.Language = LanguageNames.FSharp))
-                                            |> Array.ofSeq
-
-                                    if docIdsToReanalyze.Length > 0 then
-                                        analyzerService.Reanalyze(workspace, documentIds = docIdsToReanalyze)
-                            with ex ->
-                                logException ex
-                        }
-                        |> Async.Start // fire and forget
+                        if document.Project.Language <> LanguageNames.FSharp then
+                            ()
+                        else
+                            cancellableTask {
+                                try
+                                    if document.Project.Language = LanguageNames.FSharp then
+                                        let openDocIds = workspace.GetOpenDocumentIds()
+
+                                        let docIdsToReanalyze =
+                                            if document.IsFSharpScript then
+                                                [| for x in openDocIds do
+                                                    if x <> document.Id 
+                                                        && (let doc = solution.GetDocument(x)
+                                                            match doc with
+                                                            | null -> false
+                                                            | _ -> doc.IsFSharpScript) then 
+                                                                yield x |]
+                                            else
+                                                let depProjIds = document.Project.GetDependentProjectIds().Add(document.Project.Id)
+
+                                                [| for x in openDocIds do
+                                                    if depProjIds.Contains(x.ProjectId)
+                                                        && x <> document.Id
+                                                        && (let doc = solution.GetDocument(x)
+                                                            match box doc with
+                                                            | null -> false
+                                                            | _ -> doc.Project.Language = LanguageNames.FSharp) then
```

Comment:
This then probably also should be replaced to `true`

Reply:
No, it's different document every time.

Reply:
Oops, didn't look at the right place, ok